crates/lib/src/name.rs

```diff
@@ -287,69 +266,56 @@ impl From<Tld> for DomainPart {
 /// Each segment in a database name can contain any UTF-8 character, except for
 /// whitespace and '/'. The maximum segment length is 64 characters.
 ///
-/// [`DomainName`]s consist of [`DomainPart`]s, and are compared case-
-/// insensitively. Note, however, that the [`fmt::Display`] and [`Serialize`]
-/// impls will use the original, (potentially) mixed-case representation.
+/// Note that [`PartialEq`] compares the exact string representation of a
+/// [`DomainName`], as one would expect, but the SpacetimeDB registry compares
```

Comment:
At some point, there was a -- somewhat handwavy -- desire to "compare database names lowercase", presumably to avoid visually similar names. Since all of UTF-8 (except whitespace) is allowed in a `DomainName`, this is actually non-trivial: do we convert to Unicode normal form (and if so, which)? Do we apply proper case-folding? Or compare character-wise? Or just `to_lowercase`, which respects _some_ rules regarding the positions of characters within a word? Does that catch everything "visually similar", and what is the threat model actually?

On the other hand, it seems quite desirable to preserve the exact casing as given by the user for display purposes. 

Originally, the idea was to make `PartialEq` canonical for both a "plain" value and a SpacetimeDB column value. But alas, this doesn't work because we don't have custom comparators (yet)! 

So we need to store both the original value and a `String` key in exactly the form so `PartialEq` for `String` yields the desired comparison. Before this change, this was exactly what was happening, at least doubling the size of `DomainName` when the lowercase version was not needed most of the time.

**tl;dr** A `DomainName` is just a fancy string most of the time, except for the SpacetimeDB DNS. Because Unicode is hard, we may want to change the semantics of the DNS down the road -- but we definitely **don't** want `PartialEq` and DNS to diverge! 

If you don't mind, I would just remove the commentary here. It is, if you will, an "implementation detail" of the SpacetimeDB DNS.